src/logics/ArithLogic.cc

```diff
@@ -227,9 +250,15 @@ pair<PTRef, PTRef> ArithLogic::splitTermToVarAndConst(PTRef term) const {
         assert(getPterm(term).size() == 2);
         PTRef fac = getPterm(term)[0];
         PTRef var = getPterm(term)[1];
-        if (not isConstant(fac)) { std::swap(fac, var); }
+        if (not isConstant(fac)) {
+            std::swap(fac, var);
+            if (not isConstant(fac)) {
+                PTRef fac = yieldsSortInt(term) ? getTerm_IntOne() : getTerm_RealOne();
+                return {term, fac};
```

Comment:
I am pretty sure this is not correct.
For a term `x*y` this would return a pair `(x,1)`.

Reply:
nope, it would not
We return `{term, fac}`, and `term` is already a multiplication of its own
So it would return `{x*y, 1}`

Reply:
You are right, I overlooked that you return `term` and not `var`.